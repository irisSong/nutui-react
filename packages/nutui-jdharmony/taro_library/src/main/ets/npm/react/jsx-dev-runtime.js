'use strict';

if ("production" === 'production') {
  module.exports = require('./cjs/react-jsx-dev-runtime.production.min.js');
} else {
  module.exports = require('./cjs/react-jsx-dev-runtime.development.js');
}
