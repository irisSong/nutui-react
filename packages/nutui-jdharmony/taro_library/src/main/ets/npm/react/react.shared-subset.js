'use strict';

if ("production" === 'production') {
  module.exports = require('./cjs/react.shared-subset.production.min.js');
} else {
  module.exports = require('./cjs/react.shared-subset.development.js');
}
