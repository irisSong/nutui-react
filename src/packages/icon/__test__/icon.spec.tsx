import * as React from 'react'
import { IconFont as Icon } from '@nutui/icons-react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

test('should icon snapshot match', () => {
  const { asFragment } = render(
    <Icon className="test" name="JD" color="#FF0F23" size="24" />
  )
  expect(asFragment()).toMatchSnapshot()
})

test('should color #FF0F23', async () => {
  const { container } = render(
    <Icon className="test" name="JD" color="#FF0F23" size="24" />
  )
  expect(container.querySelector('i')?.style.color).toBeTruthy()
})

test('should classname test', async () => {
  const { container } = render(
    <Icon className="test" name="JD" color="#FF0F23" size="24" />
  )
  expect(container.querySelector('.test')).toBeInTheDocument()
})

test('should onclick called', () => {
  const handleClick = vi.fn(() => {})
  const { getByTestId } = render(
    <Icon
      data-testid="test"
      className="test"
      name="JD"
      color="#FF0F23"
      size="24"
      onClick={handleClick}
    />
  )
  fireEvent.click(getByTestId('test'))
  expect(handleClick).toBeCalled()
})
