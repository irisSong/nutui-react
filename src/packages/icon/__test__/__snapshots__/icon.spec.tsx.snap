// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should icon snapshot match 1`] = `
<DocumentFragment>
  <i
    class="nutui-iconfont nut-icon nut-icon-JD test"
    style="color: #FF0F23; font-size: 24px; width: 24px; height: 24px;"
  />
</DocumentFragment>
`;
